dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu.h

```diff
@@ -0,0 +1,234 @@
+        }
+        pitch[pitch.size() - 1] *= tv.shape[pitch.size()];
+    };
+
+    fill_pitch_with_flattening(in[i], sample.in_pitch);
+    fill_pitch_with_flattening(out[i], sample.out_pitch);
+
+    sample.brightness = brightness[i];
+    sample.contrast = contrast[i];
+  }
+
+  return ret;
+}
+
+
+template <class OutputType, class InputType, int ndims>
+__global__ void
+BrightnessContrastKernel(const SampleDescriptor<OutputType, InputType, ndims> *samples,
+                         const BlockDesc<ndims> *blocks) {
+  const auto &block = blocks[blockIdx.x];
+  const auto &sample = samples[block.sample_idx];
+
+  const auto *__restrict__ in = sample.in;
+  auto *__restrict__ out = sample.out;
+
+  for (int y = threadIdx.y + block.start.y; y < threadIdx.y + block.end.y; y += blockDim.y) {
+    for (int x = threadIdx.x + block.start.x; x < threadIdx.x + block.end.x; x += blockDim.x) {
+      out[y * sample.out_pitch.x + x] = ConvertSat<OutputType>(
+              in[y * sample.in_pitch.x + x] * sample.contrast + sample.brightness);
+    }
+  }
+}
+
+
+template <typename OutputType, typename InputType, int ndims>
+class BrightnessContrastGpu {
+ private:
+  static constexpr size_t spatial_dims = ndims - 1;
+  using BlockDesc = kernels::BlockDesc<spatial_dims>;
+
+  std::vector<SampleDescriptor<OutputType, InputType, ndims>> sample_descriptors_;
+
+ public:
+  BlockSetup<spatial_dims, -1 /* No channel dimension, only spatial */> block_setup_;
+
+
+  KernelRequirements Setup(KernelContext &context, const InListGPU<InputType, ndims> &in,
+                           const std::vector<float> &brightness, const std::vector<float> &contrast,
+                           const std::vector<Roi<spatial_dims>> &rois = {}) {
+    DALI_ENFORCE(rois.empty() || rois.size() == static_cast<size_t>(in.num_samples()),
+                 "Provide ROIs either for all or none input tensors");
+    DALI_ENFORCE([=]() -> bool {
+        for (const auto &roi : rois) {
+          if (!all_coords(roi.hi >= roi.lo))
+            return false;
+        }
+        return true;
+    }(), "One or more regions of interests are invalid");
+    DALI_ENFORCE([=]() -> bool {
+        auto ref_shape = in.shape[0][ndims - 1];
+        for (int i = 0; i < in.num_samples(); i++) {
+          if (in.shape[i][ndims - 1] != ref_shape) {
+            return false;
+          }
+        }
+        return true;
+    }(), "Number of channels for every image in batch must be equal");
+
+    auto adjusted_rois = AdjustRois(rois, in.shape);
+    auto shape = in.shape[0][ndims - 1];
+    KernelRequirements req;
+    ScratchpadEstimator se;
+    TensorListShape<spatial_dims> output_shape({RoiToShape(adjusted_rois, shape)});
+    block_setup_.SetupBlocks(output_shape, true);
+    se.add<SampleDescriptor<InputType, OutputType, ndims>>(AllocType::GPU, in.num_samples());
+    se.add<BlockDesc>(AllocType::GPU, block_setup_.Blocks().size());
+    req.output_shapes = {output_shape};
```

Comment:
Hmm, don't you lose the channel information here? You will claim that your output is flattened, the total size is ok, but not the actual shape.

Reply:
Yup, fixed it